HLTrigger/special/plugins/HLTMuonRecHitClusterFilter.cc

```diff
@@ -89,11 +110,23 @@ bool HLTMuonRecHitClusterFilter::filter(edm::StreamID, edm::Event& iEvent, const
   auto const& rechitClusters = iEvent.get(cluster_token_);
 
   for (auto const& cluster : rechitClusters) {
-    if ((cluster.size() >= min_Size_) && ((cluster.size() - cluster.nMB1()) >= min_SizeMinusMB1_) &&
-        (cluster.nMB1() <= max_nMB1_) && (cluster.nMB2() <= max_nMB2_) && (cluster.nME11() <= max_nME11_) &&
-        (cluster.nME12() <= max_nME12_) && (cluster.nME41() <= max_nME41_) && (cluster.nME42() <= max_nME42_) &&
-        (cluster.nStation() >= min_Nstation_) && (cluster.avgStation() >= min_AvgStation_) &&
-        ((min_Eta_ < 0.0) || (std::abs(cluster.eta()) >= min_Eta_)) &&
+    bool passSizeCut = false;
+    if (cluster.size() >= min_Size_)
+      passSizeCut = true;
+    if (cluster.size() - cluster.nMB1() >= min_SizeMinusMB1_)
+      passSizeCut = true;
+    for (size_t i = 0; i < min_SizeRegionCutEtas_.size(); ++i) {
+      if (((min_SizeRegionCutEtas_[i] < 0.0) || std::abs(cluster.eta()) > min_SizeRegionCutEtas_[i]) &&
+          ((max_SizeRegionCutEtas_[i] < 0.0) || std::abs(cluster.eta()) <= max_SizeRegionCutEtas_[i]) &&
+          ((min_SizeRegionCutNstations_[i] < 0) || cluster.nStation() > min_SizeRegionCutNstations_[i]) &&
+          ((max_SizeRegionCutNstations_[i] < 0) || cluster.nStation() <= max_SizeRegionCutNstations_[i]) &&
+          cluster.size() >= min_SizeRegionCutClusterSize_[i])
+        passSizeCut = true;
+    }
+    if ((passSizeCut) && (cluster.nMB1() <= max_nMB1_) && (cluster.nMB2() <= max_nMB2_) &&
+        (cluster.nME11() <= max_nME11_) && (cluster.nME12() <= max_nME12_) && (cluster.nME41() <= max_nME41_) &&
+        (cluster.nME42() <= max_nME42_) && (cluster.nStation() >= min_Nstation_) &&
+        (cluster.avgStation() >= min_AvgStation_) && ((min_Eta_ < 0.0) || (std::abs(cluster.eta()) >= min_Eta_)) &&
```

Comment:
@missirol Thanks for the comment. I cleaned up the round brackets `()` .

For the conditions, I kept `>` sign for `min*` style cuts to keep the configuration parameters look natural to users. 

 `Nstation` has the lower bound of 0, so `MinNstation`'s default value would be naturally 0, with the logic checking `>=0`.  The same goes for `Max_n*` for the vetos, because they have the lower bound of `0` and it'll be natural to have the value of `0` and checking `<=0` when the cut is active. 

The `=` sign only matters when we want to define the mutually exclusive regions like in `SizeRegionCut*`, and I chose it such that the boundary for `SizeRegionCutNstations` is `1` instead of `2`.